ocaml/xapi/message_forwarding.ml

```diff
@@ -6484,4 +6492,105 @@ functor
               ~base_build_id ~base_version ~base_release ~to_version ~to_release
         )
     end
+
+    module Tracing = struct
+      let do_trace_op_on ~__context ~self ~local_fn ~client_fn ~db_fn =
+        let localhost = Helpers.get_localhost ~__context in
+        if Helpers.is_pool_master ~__context ~host:localhost then (
+          db_fn () ;
+          (* Option.iter
+             (fun status -> Db.Tracing.set_status ~__context ~self ~value:status)
+             status ; *)
+          let hosts =
+            match Db.Tracing.get_hosts ~__context ~self with
+            | [] ->
+                Db.Host.get_all ~__context
+            | hosts ->
+                hosts
+          in
+          let hosts, local = List.partition (( <> ) localhost) hosts in
+          List.iter
+            (fun host -> do_op_on ~__context ~local_fn ~host client_fn)
+            hosts ;
+          List.iter (fun _ -> local_fn ~__context) local
+        ) else
+          local_fn ~__context ;
+        ()
+
+      let set_status ~__context ~self ~status =
+        info "Tracing.set_status: self=%s status=%B"
+          (tracing_uuid ~__context self)
+          status ;
```

Comment:
Using __FUNCTION__ makes these loglines more resilient against errors in copy-pastes, renamings and typos in general
```suggestion
        info "%s: self=%s status=%B" __FUNCTION__
          (tracing_uuid ~__context self)
          status ;
```